mlir/lib/Dialect/Linalg/Transforms/SubsetInsertionOpInterfaceImpl.cpp

```diff
@@ -9,12 +9,29 @@
 #include "mlir/Dialect/Linalg/Transforms/SubsetInsertionOpInterfaceImpl.h"
 
 #include "mlir/Dialect/Linalg/IR/Linalg.h"
-#include "mlir/Interfaces/SubsetInsertionOpInterface.h"
+#include "mlir/Interfaces/SubsetOpInterface.h"
 
 using namespace mlir;
 using namespace mlir::linalg;
 
 namespace {
+struct LinalgCopyOpSubsetOpInterface
+    : public SubsetOpInterface::ExternalModel<LinalgCopyOpSubsetOpInterface,
+                                              linalg::CopyOp> {
+  bool operatesOnEquivalentSubset(
```

Comment:
Can we add a comment here to explain?
I would have thought copy ops operate on equivalent subsets by construction.

Reply:
`false` is a conservative implementation. I added a better implementation for `operatesOnEquivalentSubset`. `linalg.copy` implements this interface only for bufferization purposes. It does not participate in subset hoisting, so `operatesOnEquivalentSubset` and `operatesOnDisjointSubset` are actually not needed at the moment.